src/main/java/com/github/tomakehurst/wiremock/matching/ContentPatternDeserialiser.java

```diff
@@ -16,30 +16,78 @@
 package com.github.tomakehurst.wiremock.matching;
 
 import com.fasterxml.jackson.core.JsonParser;
-import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.BeanProperty;
 import com.fasterxml.jackson.databind.DeserializationContext;
 import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonMappingException;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.ContextualDeserializer;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
 import java.io.IOException;
 import java.util.*;
 
-public class ContentPatternDeserialiser extends JsonDeserializer<ContentPattern<?>> {
+public class ContentPatternDeserialiser extends StdDeserializer<ContentPattern<?>>
+    implements ContextualDeserializer {
+
+  private static final Map<String, Class<? extends ContentPattern<?>>> PATTERNS =
+      new HashMap<>(
+          Map.ofEntries(
+              Map.entry("binaryEqualTo", BinaryEqualToPattern.class),
+              Map.entry("equalTo", EqualToPattern.class),
+              Map.entry("equalToJson", EqualToJsonPattern.class),
+              Map.entry("matchesJsonPath", MatchesJsonPathPattern.class),
+              Map.entry("matchesJsonSchema", MatchesJsonSchemaPattern.class),
+              Map.entry("equalToXml", EqualToXmlPattern.class),
+              Map.entry("matchesXPath", MatchesXPathPattern.class),
+              Map.entry("contains", ContainsPattern.class),
+              Map.entry("not", NotPattern.class),
+              Map.entry("doesNotContain", NegativeContainsPattern.class),
+              Map.entry("matches", RegexPattern.class),
+              Map.entry("doesNotMatch", NegativeRegexPattern.class),
+              Map.entry("before", BeforeDateTimePattern.class),
+              Map.entry("after", AfterDateTimePattern.class),
+              Map.entry("equalToDateTime", EqualToDateTimePattern.class),
+              Map.entry("anything", AnythingPattern.class),
+              Map.entry("absent", AbsentPattern.class),
+              Map.entry("and", LogicalAnd.class),
+              Map.entry("or", LogicalOr.class),
+              Map.entry("matchesPathTemplate", PathTemplatePattern.class)));
+
+  public ContentPatternDeserialiser() {
+    super(ContentPattern.class);
+  }
+
+  private ContentPatternDeserialiser(Class<?> vc) {
+    super(vc);
+  }
+
+  public static void registerPatterns(Map<String, Class<? extends ContentPattern<?>>> patterns) {
+    patterns.forEach(PATTERNS::putIfAbsent);
```

Comment:
The issue with this approach is when there are multiple WM instances running in the same JVM. Most of the time it'll probably be fine, but produce weird edge cases when not.

So far we've kept global static state out of WireMock for precisely this reason and I'd be reluctant to start introducing it here.

From my analysis a while back, there are 2 other approaches that could be made to work per-instance:
1) Make `Json` a service that's injected everywhere it's needed, then custom serializers can be attached to the `ObjectMapper` it wraps. This would be the cleanest way to do it but is a pretty large amount of work.
2) Deserialize unknown matcher keys into a `Map` (e.g. via Jackson's `@JsonAnySetter` /  `@JsonAnyGetter`) then later associate them with matcher extensions that have been registered.

Reply:
I have just pushed some code for another option:
3. Use thread local state instead of global static state. Before performing any operations that deserializes a `StubMapping` or `StubMappingCollection`, use the `Extensions` of the currently running WM instance to supply `ContentPatternDeserialiser` with the extensions that have been registered.

Reply:
1. Seems too complex of a change.
2. I may spend some time looking at this option, but I think it will look a lot like my current code. Where I make changes in the locations that deserialize `StubMapping` and `StubMappingCollection`, but instead of tampering with `ContentPatternDeserialiser`, I would post process the object to create concrete stub mappings from the generic map.